### ISSUE TITLE
Add descriptive text on Panel mode option for View Configuration UI

### ISSUE THREAD

**Issue by SeanPM5** [home-assistant/home-assistant-polymer#3835]
**Home Assistant release with the issue:**
0.99.3

**Description of problem:**
The UI editor for View Configuration has a toggle for "Panel Mode?" but doesn't actually explain what this does. So what happens is the user does one of the following:

1. Has to search through documentation to find out what this does. If the user has to Google to find out what a UI option does, it's not quite simple enough yet.
2. User toggles it on just to see what it does, and then it's such a drastic change compared to what they're normally used to. Orange plus button is gone, the edit button on the card is gone, only a single card is showing up. Were all my cards just deleted? Oh shit, panic time!

And another issue is that the toggle switch for panel mode and the delete view icons are kind of close to each other, especially for mobile devices where it has to be finger-friendly.

Putting descriptive text underneath the option would solve all these issues. Here is a very quick mockup with the text just copied and pasted from [Views documentation](https://www.home-assistant.io/lovelace/views/#panel-mode). The wording can be refined further and also styled with secondary text color etc.

![panel-mode](https://i.imgur.com/C1A6uqR.png)

**Comment by iantrich** [comment]
I think a general help icon on items or the card itself would be better that allows you to bring up more context. If we start putting in these long descriptions inline, things will get cluttered quickly

**Comment by SeanPM5** [comment]
I think a help icon is a good idea, but for this particular screen I think panel mode is the only thing that truly needs an explanation? One line of descriptive text wouldn't add too much clutter imo.

Another idea I had was moving both this "Panel Mode" toggle and "Theme" dropdown to a new Advanced tab. So the View Configuration tabs would be settings, badges, advanced. The advanced tab could also be where the eventual UI editor for user visibility settings in https://github.com/home-assistant/home-assistant-polymer/pull/3811 goes too.

**Comment by bramkragten** [comment]
I think it would be good to have some explanation in the editors, a lot of the options are not self-explanatory, especially for new users.

A help icon can work if things need even more info and can show all the options, but the basics should be clear without extra actions. 

I don't think it will clutter the UI if we just keep it as simple paragraphs with enough spacing etc.

**Comment by iantrich** [comment]
Adding a bunch of paragraphs all over our editors will absolutely clutter; they are already cluttered for many cards. I'm not in favor of this idea.

**Comment by bramkragten** [comment]
And we should remove the cluttering there is already, they should be simple and only with the basic functions. All other functions should be moved to advanced mode.

Tabs might also help to make the editors cleaner, as there is a lot going on now with some editors.

We should aim to have an UI that does not require documentation and that is self-explanatory, also for our parents.

**Comment by SeanPM5** [comment]
I think it's important to strike the right balance. Descriptive text and helper icons should only be used as an absolute last resort, the goal should always be to do without them if possible.

For the card editors specifically, there's a bunch of ideas that can be explored to clean them up and add clarity without needing text in many cases. Just a few smaller examples off the top of my head:
* If we know the user can only select certain types of things, why not have the dropdown label say "Select light" instead of a generic "Entity" label? (applies to alarm panel, light, media, thermostat, etc).
* If the user only has default theme (probably the case majority of the time), could we hide the pointless "theme" dropdown for them? 
* Entity is often the only required option, yet it's usually listed underneath or alongside a bunch of optional fields. For example on the Light card it's sandwiched between two optional settings, and is given less priority than the optional name override.


But back to this specific case for this panel mode toggle, I can't think of any better alternative options? 

IMO it would feel a little silly to have a button that you have to press just to show one sentence of descriptive text. And having a help button link off to the docs wouldn't be ideal either cause those are like 99% YAML config stuff and mostly irrelevant to UI editor users.

**Comment by iantrich** [comment]
I think we should go with @bramkragten's idea to start offloading anything not required to an Advanced area/section/tab/dialog/something and get those things out of the way to clean things up a lot.

**Comment by SeanPM5** [comment]
I think the tab idea presents some other potential issues to take into consideration. For example would a "Show icon" toggle be a basic option on the main tab and then "Icon Height" be offloaded to advanced tab? That would feel extremely awkward when clearly related options are spread across multiple tabs rather than being grouped together.

I like your line of thinking with more descriptive config name, that would be my preferred solution if we can come up with something short and clear.  "Single Card Mode" is an improvement, but still not entirely clear imo, user could think that it merges all of their cards into one single mega-card or something rather than just showing the first card.

**Comment by iantrich** [comment]
I would probably move everything besides the required fields to the advanced area.

Along these same lines, if we go this route, I think we should remove all side-by-side potions of the editors and just have a list and if one option spawns other options, it should be in a contained view for editing e.g. actions, entity options, gauge severity; much like mobile app settings
